### PR TITLE
[APB-4417]: Fix Email for CGT and Refactor Tests

### PR DESCRIPTION
**Pull request by JTY009**

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientauthorisation/service/ClientNameService.scala [hunk 1]
```diff
@@ -77,4 +78,16 @@ class ClientNameService @Inject()(
         Logger.warn(s"error during retrieving trust name for utr: ${utr.value} , error: $invalidTrust")
         None
     }
+
+  def getCgtName(cgtRef: CgtRef)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[String]] =
+    desConnector.getCgtSubscription(cgtRef).map(_.response).map {
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentclientauthorisation/service/ClientNameService.scala]: can we get this from the `cgtSubscriptionCache` ?

File: it/uk/gov/hmrc/agentclientauthorisation/support/TestDataSupport.scala [hunk 2]
```diff
@@ -45,4 +49,27 @@ trait TestDataSupport {
...
+                         urlIdentifier: String,
+                         clientId: TaxIdentifier,
+                         suppliedClientId: TaxIdentifier,
+                         wrongIdentifier: TaxIdentifier)
+
+  val itsaClient = TestClient(personal, "Trade Pears", MtdIt, MtdItIdType, "MTDITID", mtdItId, nino, mtdItId2)
+  val irvClient = TestClient(personal, "John Smith", PersonalIncomeRecord, NinoType, "NI", nino, nino, nino2)
+  val vatClient = TestClient(personal, "GDT", Vat, VrnType, "VRN", vrn, vrn, vrn2)
+  val trustClient = TestClient(business, "Nelson James Trust", Trust, UtrType, "UTR", utr, utr, utr2)
+  val cgtClient = TestClient(business, "firstName lastName", Service.CapitalGains, CgtRefType, "CGTPDRef", cgtRef, cgtRef, cgtRef2)
```
Comment (sureshhmrc) [inline it/uk/gov/hmrc/agentclientauthorisation/support/TestDataSupport.scala]: if it's a business CGT, then we only have the org name not first/last names

Comment (JTY009) [reply it/uk/gov/hmrc/agentclientauthorisation/support/TestDataSupport.scala]: Ah that's true hmmm....

File: app/uk/gov/hmrc/agentclientauthorisation/service/ClientNameService.scala (hunk 1 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentclientauthorisation/service/ClientNameService.scala]: true

File: conf/application.conf [hunk 3]
```diff
@@ -20,6 +20,9 @@ include "backend.conf"
 appName = agent-client-authorisation
 # Define any modules used here
 
+# Override for Local Development
+logger.root = INFO
```
Comment (sureshhmrc) [inline conf/application.conf]: is this overriden in production?

**Pull request merged**